src/js/profiler.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+/**
+ * @class Snapshot
+ */
+function Snapshot() {
+  // native.takeSnapshot();
+  throw new Error('not implemented');
+}
+
+/**
+ * @method getHeader
+ */
+Snapshot.prototype.getHeader = function() {
+  // TODO
+  throw new Error('not implemented');
+};
+
+/**
+ * @method compare
+ */
+Snapshot.prototype.compare = function(snapshot) {
+  // TODO
+  throw new Error('not implemented');
+};
+
+/**
+ * @method export
+ */
+Snapshot.prototype.export = function(cb) {
+  // TODO
+  throw new Error('not implemented');
+};
+
+/**
+ * @method serialize
+ */
+Snapshot.prototype.serialize = function() {
+  // TODO
+  throw new Error('not implemented');
+};
+
+/**
+ * @class Profile
+ */
+function Profile() {
+  // TODO
+}
+
+/**
+ * @method getHeader
+ */
+Profile.prototype.getHeader = function() {
+  throw new Error('not implemented');
+};
+
+/**
+ * @method delete
+ */
+Profile.prototype.delete = function() {
+  throw new Error('not implemented');
+};
+
+/**
+ * @method export
+ */
+Profile.prototype.export = function(cb) {
+  throw new Error('not implemented');
+};
+
+/**
+ * @method getHeader
+ */
+function takeSnapshot() {
+  return new Snapshot();
+}
+
+/**
+ * startProfiling(path)
+ * startProfiling(duration)
+ * startProfiling(path, duration)
+ */
+function startProfiling() {
+  var path = `${process.cwd()}/Profile-${Date.now()}`;
+  var duration = -1;
+
+  if (arguments.length === 1) {
+    if (typeof (arguments[0]) === 'number') {
+      duration = arguments[0];
+    } else if (typeof (arguments[0]) === 'string') {
+      path = arguments[0];
+    }
+  } else if (arguments.length === 2) {
+    path = arguments[0];
+    duration = arguments[1];
+  }
+
+  native.startProfiling(path, duration);
+}
+
+function stopProfiling() {
+  native.stopProfiling();
```

Comment:
Suggestion: use `process._readParserDump()` here to make parser dump in persistent, such that can remove [these lines](https://github.com/Rokid/ShadowNode/blob/0e91917c138b1b5dfd5505f691524ee58190dddf/deps/jerry/jerry-core/api/jerry.c#L3581-L3587).